aspnet-core/src/AbpCompanyName.AbpProjectName.Web.Host/Startup/SecurityRequirementsOperationFilter.cs

```diff
@@ -10,22 +10,28 @@ public class SecurityRequirementsOperationFilter : IOperationFilter
     {
         public void Apply(Operation operation, OperationFilterContext context)
         {
-            var controllerPermissions = context.ApiDescription.ControllerAttributes()
-                .OfType<AbpAuthorizeAttribute>()
-                .Select(attr => attr.Permissions);
-
-            var actionPermissions = context.ApiDescription.ActionAttributes()
-                .OfType<AbpAuthorizeAttribute>()
-                .Select(attr => attr.Permissions);
+            var actionAttrs = context.ApiDescription.ActionAttributes();
+            if (actionAttrs.OfType<AbpAllowAnonymousAttribute>().Any())
```

Comment:
`AbpAllowAnonymousAttribute` can be on controller too, right?

Reply:
@acjh yes 😄

Reply:
Attributes on action have highest priority. AbpAuthorizeAttribute on action will cover AbpAllowAnonymousAttribute  on controller. Am I right?

Reply:
If by *cover*, you mean *take priority over*, then yes.

```c#
if (!actionAbpAuthorizeAttrs.Any() && controllerAttrs.OfType<AbpAllowAnonymousAttribute>().Any())
{
    return;
}
```

Reply:
Without global Authorization filters, this condition is embodied by:
```c#
if (controllerAbpAuthorizeAttrs.Any() || actionAbpAuthorizeAttrs.Any())
{
}
```
I don't think that global Authorization filters is a good idea on ABP.
And, I will take your advise. Anyway, it does not affect the performance of the application.